src/NuGet.Core/NuGet.Commands/RestoreCommand/OriginalCaseGlobalPackageFolder.cs

```diff
@@ -27,7 +27,14 @@ public class OriginalCaseGlobalPackageFolder
         private readonly ToolPathResolver _toolPathResolver;
         private readonly VersionFolderPathResolver _pathResolver;
 
-        public OriginalCaseGlobalPackageFolder(RestoreRequest request)
+        public KeyValuePair<string, Guid> ParentId { get; }
```

Comment:
similar to operation id, why do you need a kvp for parent id?

Reply:
because you can define different key for parentId, for example, if we just use parentId, then you have to understand the whole parent-child logic to understand how to join the parentId
By using kvp, you can define "restoreId" in restoreCommand, "extractionId" in extraction, then you can find the parent by id name easily.